### ISSUE TITLE
SQL ILIKE will fails on string with \n inside

### ISSUE THREAD

**Issue by ExtReMLapin** [ArcadeData/arcadedb#1577]
Hello, it's me again. TL;DR at the bottom

### ArcadeDB Version:

`ArcadeDB Server v24.5.1-SNAPSHOT (build 71472c4c05ae9278d0d5d71a919b31883e1ea767/1713899543605/main) is starting up...` creating nodes and reading nodes

### OS and JDK Version:

`Running on Linux 5.15.0-100-generic - OpenJDK 64-Bit Server VM 11.0.22` for creating nodes and reading nodes

### Expected behavior

ILIKE should match in any case not based on existence of `\n` inside the string

### Actual behavior

ILIKE fails when there is any `\n` inside the string.

### Steps to reproduce

Pick up this database : 

[test_db.zip](https://github.com/ArcadeData/arcadedb/files/15150404/test_db.zip)

Try both : 

```SQL
SELECT *  FROM CHUNK"
```

```SQL
SELECT *  FROM CHUNK where text ILIKE "%e%"
```

In theory, both request should output same result as all the CHUNK nodes have at least one `e` in the `text` field, however, here it fails to catch one of them.

## TLDR : 

All 4 have `e`, ILIKE should match all of them, but it doesn't.

![firefox_xuuEyHHDPs](https://github.com/ArcadeData/arcadedb/assets/3909752/8c16f9ec-737d-439b-83ff-dcab592886a3)
![firefox_GfRXRz83d5](https://github.com/ArcadeData/arcadedb/assets/3909752/9c3b536a-42f6-4e2d-8de4-02ddab1b0574)

This is the correct query to execute as a temp fix :  

```SQL
SELECT *  FROM CHUNK where text.replace('\n', '') ILIKE "%e%"
```

Thanks for the help.

**Comment by ExtReMLapin** [comment]
What's weird is that when created on another database, isolated, there is no issue.

![image](https://github.com/ArcadeData/arcadedb/assets/3909752/0f12f61d-1aac-4a7b-92fd-5d5c8a18a714)

**Comment by gramian** [comment]
Hi,
does the same happen with `LIKE` + `toLowerCase()`, ie:
```
SELECT *  FROM CHUNK where text.toLowerCase() LIKE "%e%"
```

**Comment by ExtReMLapin** [comment]
Yep, only 3 records, still not 4

**Comment by ExtReMLapin** [comment]
It also could be something than the text len, it could be a special character in the string as spamming `e` didn't cause this issue

**Comment by gramian** [comment]
I was also thinking of special chars, could you try `normalize()` in the filter, ie:
```
SELECT *  FROM CHUNK where text.normalize() ILIKE "%e%"
```

**Comment by ExtReMLapin** [comment]
It's the `\n` that breaks it.

**Comment by ExtReMLapin** [comment]
Sorry for the spam, but `SELECT *  FROM CHUNK where text.replace('\n', '') ILIKE "%e%"` fixes it

**Comment by gramian** [comment]
Good find. This may still be an issue or at least needs documentation.

**Comment by gramian** [comment]
@ExtReMLapin this should be fixed in the current repo head. Feel free to reopen if you find a problem.
